src/vsr/replica.zig

```diff
@@ -9434,6 +9843,20 @@ const PipelineQueue = struct {
         return message;
     }
 
+    fn message_by_operation(pipeline: PipelineQueue, operation: vsr.Operation) ?*const Message {
```

Comment:
This feels like an error-prone API, because it is not specified _which_ message is returned in case several match. 

Maybe `fn operation_pipelined(pipeline: PipelineQueue, operation: vsr.Operation) bool`?

Internally, it'll still compute the actual message and assert that we do have a message with the specifed op, but the call site won't be able to rely on impl detail of which message is returned.